examples/llama.py

```diff
@@ -452,37 +452,23 @@ def main(string: str):
   llama = LLaMa.build(MODEL_PATH, TOKENIZER_PATH, model_gen=args.gen, model_size=args.size, quantize=args.quantize, device=device)
   param_bytes = sum(x.lazydata.size * x.dtype.itemsize for x in get_parameters(llama.model))
 
+  outputted = pre_prompt if chatbot else args.prompt
+  start_pos, toks = 0, [llama.tokenizer.bos_id()] + llama.tokenizer.encode(outputted)
   if chatbot:
-    # encode pre prompt
-    toks = [llama.tokenizer.bos_id()] + llama.tokenizer.encode(pre_prompt)
-
     print(f"Preparing KV cache for chatbot with personality {args.personality}...")
+    start_pos = len(toks)
     with Timing():
       llama.model(Tensor([toks], device=device), 0, args.temperature).realize()  # NOTE: outputs are not used
-    start_pos = len(toks)
-  else:
-    # non chat bot mode
-    toks = [llama.tokenizer.bos_id()] + llama.tokenizer.encode(args.prompt)
-    start_pos = 0
-
-  # print prompt
-  outputted = llama.tokenizer.decode(toks)
-  sys.stdout.write(outputted)
-  sys.stdout.flush()
+  print(outputted, end='', flush=True)
 
   # chatbot loop
   while 1:
     # add tokens from user in chatbot mode
     if chatbot:
       user_prompt = user_delim + input(user_delim) + "\n"
       outputted += user_prompt
+      toks.extend(llama.tokenizer.encode(user_prompt))
 
-    new_toks = [llama.tokenizer.bos_id()] + llama.tokenizer.encode(outputted)
-    assert toks == new_toks[:len(toks)]
```

Comment:
This assert doesn't hold anymore, for example `]\n` encodes to 933 but `]\n\n` is `2595` so concating `]\n` and `\n` changes the result.